src/core/config/openthread-core-config-check.h

```diff
@@ -93,6 +93,10 @@
 
 //---------------------------------------------------------------------------------------------------------------------
 
+#ifdef OPENTHREAD_CONFIG_HEAP_EXTERNAL_ENABLE
+#error "OPENTHREAD_CONFIG_HEAP_EXTERNAL_ENABLE was replaced by OPENTHREAD_CONFIG_EXTERNAL_HEAP_ENABLE"
```

Comment:
Why are we renaming this? We have other configs that share the same `OPENTHREAD_CONFIG_HEAP_` prefix.

Reply:
Not really necessary, reverted the change.